src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -831,10 +854,14 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
+            });
+            // TODO: this should be returning S_ASYNC_STOP
```

Comment:
1. _requestingRealAsyncBreak should change to a tristate with reasons of AsyncBreak, AD7Stop, or InternalBreak.
2. In the case of stop, we should examine the MI result to determine if GDB is already stopped - in which case we should return S_OK, or if it will stop in the future, in which case we should return S_ASYNC_STOP
3. The code that is checking _requestingRealAsyncBreak should raise a stop complete event if the tristate is in 'AD7Stop' mode.